File: backend/app/schemas/alert.py
"""Alert schemas."""

from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class AlertSeverity(str, Enum):
    """Supported alert severity levels."""

    INFO = "INFO"
    WARNING = "WARNING"
    CRITICAL = "CRITICAL"


class AlertType(str, Enum):
    """Supported alert types."""

    HIGH_ENGINE_TEMP = "HIGH_ENGINE_TEMP"
    LOW_FUEL = "LOW_FUEL"
    LOW_BATTERY = "LOW_BATTERY"
    LOW_OIL_LIFE = "LOW_OIL_LIFE"
    LOW_COOLANT = "LOW_COOLANT"
    LOW_TIRE_PRESSURE = "LOW_TIRE_PRESSURE"
    HIGH_BRAKE_WEAR = "HIGH_BRAKE_WEAR"
    ENGINE_OVERLOAD = "ENGINE_OVERLOAD"
    VEHICLE_HEALTH_CRITICAL = "VEHICLE_HEALTH_CRITICAL"


class Alert(BaseModel):
    """Single vehicle alert."""

    severity: AlertSeverity
    type: AlertType
    title: str
    description: str
    recommendation: str
    timestamp: datetime


class AlertReport(BaseModel):
    """Collection of generated vehicle alerts."""

    alert_count: int = Field(ge=0)
    highest_severity: AlertSeverity | None
    alerts: list[Alert]

File: frontend/src/types/alert.ts
export type AlertSeverity = "INFO" | "WARNING" | "CRITICAL";

export type AlertType =
  | "HIGH_ENGINE_TEMP"
  | "LOW_FUEL"
  | "LOW_BATTERY"
  | "LOW_OIL_LIFE"
  | "LOW_COOLANT"
  | "LOW_TIRE_PRESSURE"
  | "HIGH_BRAKE_WEAR"
  | "ENGINE_OVERLOAD"
  | "VEHICLE_HEALTH_CRITICAL";

export interface Alert {
  severity: AlertSeverity;
  type: AlertType;
  title: string;
  description: string;
  recommendation: string;
  timestamp: string;
}

export interface AlertReport {
  alert_count: number;
  highest_severity: AlertSeverity | null;
  alerts: Alert[];
}
